client/src/lib/data.ts

```diff
@@ -3,6 +3,24 @@ import { PostComment, Comment } from '../pages/Details';
 
 export const tokenKey = 'um.token';
 
+export type Recipe = {
+  recipeId: string;
+  title: string;
+  userId: string;
+  imageUrl: string;
+  preparationTime: string;
+  cuisine: string;
+  glutenFree: boolean;
+  vegetarian: boolean;
+  ingredients: string;
+  instructions: string;
+  createdAt: string;
+};
+
+export type Recipe1 = Recipe & {
```

Comment:
The word `Recipe1` doesn't contain much information. I'm not sure what it is used for. Since it adds a username, I assume it is a recipe that someone has created? Perhaps `UserRecipe` would be a better name??